src/awaiter.h

```diff
@@ -251,59 +252,31 @@ class FutureAwaiter final: public FuturePollEvent {
   void traceEvent(kj::_::TraceBuilder& builder) override {
     // Just defer to our enclosing Coroutine. It will immediately call our CoAwaitWaker's
     // `tracePromise()` implementation.
-    static_cast<Event&>(coroutine).traceEvent(builder);
+    onReadyEvent.traceEvent(builder);
+  }
+
+  void onReady(kj::_::Event* event) noexcept override {
+    onReadyEvent.init(event);
+    fire();
```

Comment:
- make awaitSuspendImpl call onREadyEvent.arm
- call awaitSuspendImpl directly from ready
- do not call fire